src/main/java/com/easypost/model/Report.java

```diff
@@ -42,32 +41,21 @@ public static Report create(final Map<String, Object> params) throws EasyPostExc
      */
     public static Report create(final Map<String, Object> params, final String apiKey) throws EasyPostException {
         String type = (String) params.get("type");
-        String url = reportURL(type) + "?";
+
+        Map<String, Object> wrappedParams = new HashMap<String, Object>();
+        wrappedParams.put("report", params);
+        wrappedParams.put("start_date", params.get("start_date"));
+        wrappedParams.put("end_date", params.get("end_date"));
 
         if (params.containsKey("columns")) {
-            List<String> columns = (ArrayList<String>) params.get("columns");
-            for (int i = 0; i < columns.size(); i++) {
-                url += "columns[]=" + columns.get(i) + "&";
-            }
-            // Delete the columns from the param since it's added to the url query.
-            params.remove("columns");
+            wrappedParams.put("columns", params.getOrDefault("columns", new ArrayList<String>()));
         }
 
         if (params.containsKey("additional_columns")) {
-            List<String> additionalColumns = (ArrayList<String>) params.get("additional_columns");
-            for (int i = 0; i < additionalColumns.size(); i++) {
-                url += "additional_columns[]=" + additionalColumns.get(i) + "&";
-            }
-            // Delete the additional columns from the param since it's added to the url query.
-            params.remove("additional_columns");
+            wrappedParams.put("additional_columns", params.getOrDefault("additional_columns", new ArrayList<String>()));
```

Comment:
We only step into this line if `additional_columns` exists as a key in `params`, so planning for a default seems redundant.

Instead, let's just do:
```java
wrappedParams.put("additional_columns", params.getOrDefault("additional_columns", new ArrayList<String>()));
```
below `end_date` on line 48, and get rid of this if statement. Same for `columns`

In fact, we don't even need to add pretty much all the params from `params` to `wrappedParams`. We could just remove `type` from params and then use the `params` object as is.